velox/exec/Task.cpp

```diff
@@ -2833,6 +2836,66 @@ std::shared_ptr<ExchangeClient> Task::getExchangeClientLocked(
   return exchangeClients_[pipelineId];
 }
 
+std::optional<trace::QueryTraceConfig> Task::maybeMakeTraceConfig() {
+  const auto& queryConfig = queryCtx_->queryConfig();
+  if (!queryConfig.queryTraceEnabled()) {
+    return std::nullopt;
+  }
+
+  traceDirectory_ = queryConfig.queryTraceDir();
+  VELOX_CHECK(
```

Comment:
Can you add a check ahead that traceDirectory_ is empty? Thanks!

Reply:
removed `traceDirectory_`.